src/utils/book-api.js

```diff
@@ -0,0 +1,43 @@
+import {useEffect, useState} from 'react';
+
+const books = {};
+
+export async function getBookData(isbn) {
+    if (books[isbn]) return books[isbn];
+    const searchParams = new URLSearchParams();
+    searchParams.set('bibkeys', `ISBN:${isbn}`);
+    searchParams.set('jscmd', 'data');
+    searchParams.set('format', 'json');
+    const response = await fetch(`/api/books?${searchParams}`, {
+        headers: {
+            'accept': 'application/json'
+        }
+    });
+    if (response.ok) {
+        const body = await response.json();
+        books[isbn] = body[`ISBN:${isbn}`];
+        return books[isbn];
+    }
+    if (response.status === 404) {
+        return '';
```

Comment:
It might be confusing that we are returning an object most of the time but returning an empty string here

Reply:
Agreed, this is weird. I am going to change this such that it returns an empty object.